### PR TITLE
jsonutils.toJson now serializes JsonNode as is by default

### PR DESCRIPTION
**Pull request by timotheecour**

### CODE REVIEW

**Comment by Varriount** [comment]
Why special-case JsonNode? Why not treat it as-is? I feel that this adds unnecessary implementation complexity for a rare situation.

**Comment by timotheecour** [comment]
Why special-case JsonNode? Why not treat it as-is? 

treating it as is means special-casing JsonNode, actually. Otherwise it'd use the generic code path for ref.

std/json does the same: `template `%`*(j: JsonNode): JsonNode = j`, but even if it didn't, this is the right default, for the same reason Table+friends have custom handlers instead of using the generic implementation for ref|object|etc

```nim
import std/[json,jsonutils]
type A = object
  x1: JsonNode
  x2: string
let a = A(x1: %[1,2], x2: "ab")
echo %a
echo a.toJson
```
* before PR:
{"x1":[1,2],"x2":"ab"}
{"x1":{"isUnquoted":false,"kind":6,"elems":[{"isUnquoted":false,"kind":2,"num":1},{"isUnquoted":false,"kind":2,"num":2}]},"x2":"ab"}
* after PR:
{"x1":[1,2],"x2":"ab"}
{"x1":[1,2],"x2":"ab"}

> I feel that this adds unnecessary implementation complexity for a rare situation.

the `jsonutils.toJson` implementation is a straightforward switch on the type, nothing complicated there. And having JsonNode objects is not a rare situation, it's actually pretty common. As for why have `opt.jsonNodeMode`, well, that's a design goal of jsonutils, it has sane defaults but allows reasonable customizations, eg missing/extra fields, policy for enums, chars, hooks.

**Comment by Varriount** [comment]
> treating it as is means special-casing JsonNode, actually. Otherwise it'd use the generic code path for ref.

Let me rephrase: Why shouldn't a `JsonNode` field be treated as a regular object?

Additionally, this would appear to violate the rule that serialized data* should always be deserializable into the same form. If a user serializes a structure containing a JsonNode, will they be able to unserialize the result to get the previous value?
```
import tables, json, std/jsonutils

type Foo = object
  number: int
  meta: JsonNode
  
var foo = Foo(
  number: 1,
  meta: parseJson("1")
)

var fooJson = toJson(foo)

assert jsonTo(toJson(foo), Foo) == foo
```
(I'm aware that the assertion technically fails. I consider that yet another design flaw)


*excepting comments

**Comment by timotheecour** [comment]
> Let me rephrase: Why shouldn't a JsonNode field be treated as a regular object?

because special casing JsonNode is a better default:
* that's also what `json.%` does
* it gives more compact output
* jsonutils and json also specialize for other common types (Table etc), for good reasons
* it allows having objects that already contain serialized data as fields
* it allows roundtrip

> Additionally, this would appear to violate the rule that serialized data* should always be deserializable into the same form. If a user serializes a structure containing a JsonNode, will they be able to unserialize the result to get the previous value?
> (I'm aware that the assertion technically fails. I consider that yet another design flaw)

did you actually try this PR? the assert passes after this PR (and this PR has a test for that).
not sure what other design flaws you're referring to.
